pkg/server/endpoints/endpoints.go

```diff
@@ -1,27 +1,28 @@
 package endpoints
 
 import (
-	"crypto/ecdsa"
+	"crypto"
 	"crypto/tls"
 	"crypto/x509"
 	"errors"
 	"fmt"
 	"net"
 	"os"
-	"sync"
 
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 
 	"github.com/spiffe/spire/pkg/common/auth"
+	"github.com/spiffe/spire/pkg/common/bundleutil"
 	"github.com/spiffe/spire/pkg/common/peertracker"
 	"github.com/spiffe/spire/pkg/common/util"
 	"github.com/spiffe/spire/pkg/server/endpoints/node"
 	"github.com/spiffe/spire/pkg/server/endpoints/registration"
-	"github.com/spiffe/spire/pkg/server/svid"
+	"github.com/spiffe/spire/pkg/server/federation"
```

Comment:
As a passing observation, from an organization perspective, it is confusing that I can't find the bundle endpoint server code inside `github.com/spiffe/spire/pkg/server/endpoints`.

The code we are writing here is effectively a "bundle endpoint" server and client, and the spec it implements is the SPIFFE Trust Domain and Bundle spec. So, maybe it is to say that if you use the bundle endpoint server and client, you _get_ federation? Not trying to be too nitpicky here, but I think that our choice in terminology will be important going forward. Curious to hear your thoughts on the naming as well as the code organization